test/test-connector.js

```diff
@@ -212,14 +212,36 @@ describe('swagger connector', function() {
         done();
       });
     });
+
+    it('supports custom swagger client', done => {
+      const customSwaggerClient = {
+        execute: function(requestObject) {
+          requestObject.on.response({id: 'custom'});
+        },
+      };
+
+      const ds = createDataSource('test/fixtures/petStore.json',
+        false, false, customSwaggerClient);
+
+      ds.on('connected', () => {
+        const PetService = ds.createModel('PetService', {});
+        PetService.getPetById({petId: 7}, function(err, res) {
+          if (err) return done(err);
+          should(res).containDeep({id: 'custom'});
+          done();
+        });
+      });
+    });
   });
 });
 
-function createDataSource(spec, validateSpec, remotingEnabled) {
+function createDataSource(spec, validateSpec, remotingEnabled,
+  customSwaggerClient) {
```

Comment:
The number of arguments is growing out of hand. Could you please change the signature of this function as follows?

```js
function createDataSource(spec, options) {
  const config = Object.assign({
    connector: require('../index'),
    spec: spec,
  }, options);
  return loopback.createDataSource('swagger', config);
}

// sample usage:
// createDataSource('test/fixtures/petStore.json', {swaggerClient: customSwaggerClient})
```